stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/event/EventHandlerImpl.kt

```diff
@@ -392,6 +392,12 @@ internal class EventHandlerImpl(
                     domainImpl.repos.deleteChannelMessagesBefore(event.cid, event.createdAt)
                     domainImpl.repos.setChannelDeletedAt(event.cid, event.createdAt)
                 }
+                is MessageDeletedEvent -> {
+                    if (event.hardDelete) {
+                        domainImpl.repos.deleteChannelMessage(event.message)
+                        domainImpl.repos.evictChannelMessage(event.cid, event.message.id)
+                    }
```

Comment:
And this guy is less obvious. Basically it resets 3 caches:
1. In memory cache in channel repository.
2. In-memory cache in message repository. 
3. Database cache in message repository.